aws-lc-rs/src/agreement.rs

```diff
@@ -378,6 +363,36 @@ impl EphemeralPrivateKey {
     }
 }
 
+#[cfg(test)]
+fn from_ec_private_key(priv_key: &[u8], nid: i32) -> Result<LcPtr<*mut EVP_PKEY>, Unspecified> {
+    use crate::ptr::DetachableLcPtr;
+
+    let ec_group = unsafe { ec_group_from_nid(nid)? };
+    let priv_key = DetachableLcPtr::try_from(priv_key)?;
+
+    let pkey = unsafe { ec::ec_key_from_private(&ec_group.as_const(), &priv_key.as_const())? };
+
+    Ok(pkey)
+}
+
+pub(crate) fn generate_x25519() -> Result<LcPtr<*mut EVP_PKEY>, Unspecified> {
```

Comment:
Not a blocker, but I'm curious if `generate_x25519` can reuse `ec_key_generate` as well, now that we use the same `EVP_PKEY_keygen` APIs in both?

Reply:
ec_key_generate internally also has to set the EC curve parameters on the EVP_KEY, so would still at least need some distinct setup logic. Some of it could potentially be refactored to allow passing a callback function that could set additional properties before the keygen step. I will think it over, and can include it in a separate PR if it makes sense.